CMakeLists.txt

```diff
@@ -820,6 +825,24 @@ check_symbol_exists(setsockopt     "${CURL_INCLUDES}" HAVE_SETSOCKOPT)
 # symbol exists in win32, but function does not.
 check_function_exists(inet_pton HAVE_INET_PTON)
 
+if(CURL_XATTR)
+  check_symbol_exists(fsetxattr "${CURL_INCLUDES}" HAVE_FSETXATTR)
+  if(HAVE_FSETXATTR)
+    add_definitions(-DHAVE_FSETXATTR)
```

Comment:
This adds an option to the compile command. An alternative is to define macros in a config.h file (lib/curl_config.h.cmake):

    #cmakedefine HAVE_FSETXATTR 1

I think this is closer to what autotools does via `AC_DEFINE` (but I did not check the output).